File: iteration-2/src/CourseSession.java
import java.util.List;

public class CourseSession {
    private int sessionId;
    private Course course;
    private List<Schedule> scheduleList;

    public CourseSession(int sectionId, Course course, List<Schedule> scheduleList) {
        this.sessionId = sectionId;
        this.course = course;
        this.scheduleList = scheduleList;
    }


    public int getSessionId() {return sessionId;}

    public void setSessionId(int sessionId) {this.sessionId = sessionId;}

    public Course getCourse() {return course;}

    public void setCourse(Course course) {this.course = course;}

    public List<Schedule> getScheduleList() {return scheduleList;}

    public CourseSession setScheduleList(List<Schedule> scheduleList) {
        this.scheduleList = scheduleList;
        return this;
    }

    @Override
    public String toString() {
        //This method returns a string representation of the CourseSession object
        return "CourseSession{" +
                "sessionId=" + sessionId +
                ", course=" + course +
                ", scheduleList=" + scheduleList +
                '}';
    }
}

File: iteration-3/CourseSession.py
class CourseSession:
    def __init__(self, session_id, course, schedule_list):
        self.session_id = session_id
        self.course = course
        self.schedule_list = schedule_list

    def get_session_id(self):
        return self.session_id

    def set_session_id(self, session_id):
        self.session_id = session_id

    def get_course(self):
        return self.course

    def set_course(self, course):
        self.course = course

    def get_schedule_list(self):
        return self.schedule_list

    def set_schedule_list(self, schedule_list):
        self.schedule_list = schedule_list

    def __str__(self):
        return "CourseSession{{ sessionId={}, course={}, scheduleList={} }}".format(self.session_id, self.course, self.schedule_list)